packages/client/src/pages/register.vue

```diff
@@ -1,152 +1,274 @@
-import { reactive, ref, toRaw } from "vue";
+import {
+  mdiEye,
+  mdiEyeOff,
+  mdiInformationOutline,
+} from "@quasar/extras/mdi-v7";
+import { omit } from "lodash-es";
+import { QInputProps, uid } from "quasar";
+import { computed, ref } from "vue";
 import { useI18n } from "vue-i18n";
-import PasswordStrengthBar from "components/password-strength-bar.vue";
 import SocialAuthButtons from "components/social-auth-buttons.vue";
-import { RegisterPayload } from "src/@generated/graphql";
-import KPasswordInput from "src/components/k-password-input.vue";
+import { RegisterUserPayload } from "src/@generated/graphql";
 import { STRENGTH_BAR_STEPS } from "src/components/models";
+import PasswordStrengthBar from "src/components/password-strength-bar.vue";
+import { useTheme } from "src/composables/use-theme";
 import { notifyError } from "src/helpers/error-messages";
 import {
-  emailRule,
   makeValueMatchRule,
   requiredRule,
   validatePasswordRule,
 } from "src/helpers/rules";
+import { AvailableRouteNames } from "src/models/routes";
 import { useRegisterMutation } from "src/services/auth";
 import { useRetailLocationService } from "src/services/retail-location";
 
 const SOCIAL_LOGIN_ENABLED = process.env.SOCIAL_LOGIN_ENABLED === "true";
 
-const { t, locale } = useI18n();
+const { t } = useI18n();
+
+const { theme } = useTheme();
 
 const showPassword = ref(false);
 
-const user = reactive<Omit<RegisterPayload, "retailLocationId">>({
+type UserRegistrationData = Omit<
+  RegisterUserPayload,
+  "discount" | "retailLocationId" | "notes" | "dateOfBirth"
+> & {
+  confirmEmail: string;
+  dateOfBirth: string;
+};
+
+const user = ref<UserRegistrationData>({
   email: "",
   firstname: "",
   lastname: "",
   password: "",
+  confirmEmail: "",
+  phoneNumber: "",
   passwordConfirmation: "",
+  dateOfBirth: "",
 });
 
 const { register, loading: isRegistering } = useRegisterMutation();
 
+const emailMatchRule = makeValueMatchRule(
+  () => user.value.confirmEmail,
+  () => t("auth.emailsDoNotMatch"),
+);
 const passwordMatchRule = makeValueMatchRule(
-  () => user.password,
+  () => user.value.password,
   () => t("auth.passwordDoNotMatch"),
 );
 
+const formData = computed<
+  {
+    field: keyof UserRegistrationData;
+    tooltip?: string;
+    inputData: Omit<QInputProps, "modelValue">;
+  }[]
+>(() => [
+  {
+    field: "firstname",
+    inputData: {
+      label: t("auth.firstName"),
+      rules: [requiredRule],
+    },
+  },
+  {
+    field: "lastname",
+    inputData: {
+      label: t("auth.lastName"),
+      rules: [requiredRule],
+    },
+  },
+  {
+    field: "dateOfBirth",
+    inputData: {
+      label: t("auth.birthDate"),
+      type: "date",
+      rules: [requiredRule],
+    },
+  },
+  {
+    field: "delegate",
+    inputData: {
+      label: t("auth.nameOfDelegate"),
+      rules: [
+        new Date().getUTCFullYear() -
+          new Date(user.value.dateOfBirth).getUTCFullYear() <
+        18
```

Comment:
Unfortunately, this is a little bit too optimistic 😂 because it does not consider the day in which the user was born.
For example: if we consider today `2024-05-28` as the current day, and we take somebody who was born on `2024-05-30`, according to this calculation he will be considered an adult.

---

In order to properly take into account dates here you need to do a comparison like the following one:
```ts
const birthDate = new Date(user.value.dateOfBirth);
birthDate.setYear(birthDate.getFullYear() + 18);
const isAdult = (Date.now() -
          birthDate.getTime()) > 0 ;
```

There are quite a [lot of methods](https://stackoverflow.com/questions/4060004/calculate-age-given-the-birth-date-in-the-format-yyyymmdd) in reality, and since here we do not care about precisions or hours, we can simply go with one of these or my suggestion

Reply:
An alternative solution using Quasar date utils:
```ts
const ageInDays = date.getDateDiff(new Date(), new Date(user.value.dateOfBirth), 'day')
const age = Math.trunc(ageInDays / 365)
```
not using `year` as the unit as it would produce the same result as now